### PR TITLE
add WA for AWS VMs

### PR DESCRIPTION
**Pull request by LotharKAtt**

### CODE REVIEW

File: memory.go [hunk 1]
```diff
@@ -123,6 +126,46 @@ func getStructureTableAddress(f *os.File) (address int64, length int, err error)
 	return 0, 0, ErrNotExist
 }
 
+func parseProcMemfile() map[string]uint64 {
+	m := make(map[string]uint64)
+
+	memfile, err := os.Open("/proc/meminfo")
+	if err != nil {
+		log.Fatal(err)
```
Comment (mceloud) [inline memory.go]: i would not use log.Fatal

File: memory.go [hunk 2]
```diff
@@ -123,6 +126,46 @@ func getStructureTableAddress(f *os.File) (address int64, length int, err error)
...
+		}
+
+		key := text[:n]
+		data := strings.Split(strings.Trim(text[(n+1):], " "), " ")
+
+		if len(data) == 1 {
+			value, err := strconv.ParseUint(data[0], 10, 64)
+			if err != nil {
+				continue
+			}
```
Comment (mceloud) [inline memory.go:151]: you don't set m here

File: memory.go [hunk 3]
```diff
@@ -238,4 +281,17 @@ loop:
...
 		si.Memory.Size = memSizeAlt
 	}
+
+	// WA for AWS VMs
+	output, err := exec.Command("/bin/sh", "-c", "head -c 3 /sys/devices/virtual/dmi/id/product_uuid").Output()
+	if string(output) == "ec2" {
+		procmem := parseProcMemfile()
+		mem := procmem["MemTotal"]
+		if err != nil {
+			fmt.Errorf("error during parsing of memfile")
```
Comment (mceloud) [inline memory.go]: missing log or return

File: memory.go [hunk 4]
```diff
@@ -238,4 +282,18 @@ loop:
 		si.Memory.Type = "DRAM"
 		si.Memory.Size = memSizeAlt
 	}
+
+	// WA for AWS VMs
+	output, _ := exec.Command("/bin/sh", "-c", "head -c 3 /sys/devices/virtual/dmi/id/product_uuid").Output()
+	if string(output) == "ec2" {
+		procmem, err := parseProcMeminfo()
```
Comment (pupapaik) [inline memory.go:290]:
```
if err != nil {
 return
}
```

**Pull request merged**